src/listContext.js

```diff
@@ -38,18 +43,32 @@ const ListContextProvider = props => {
     return isDupe;
   };
   const validToken = token => {
+    if (fetchList(token).length > 0) {
+      console.log('here from validToken()', token);
+
+      return true;
+    }
     let filteredByToken = shoppingList.filter(item => item.token === token);
     return filteredByToken.length > 0;
+    // return true;
   };
 
-  const addItem = (name, nextExpectedPurchase) => {
+  const addItem = (name, nextExpectedPurchase, token) => {
+    // so we can add a first item to a first list
+
     if (!isDuplicate(name)) {
-      // so we can add a first item to a first list
       if (!validToken(token)) {
         saveToken(token);
```

Comment:
The 2 lines above are confusing to me. They appear to be saying "if we do not have a valid token, save the token" ... is that correct? The right thing might be happening, but the way it's expressed doesn't sound like the behavior we want.

Reply:
If the token isn't in the database because it's for a new list, it needs to be saved to localstorage so that the correct list is added to.

Reply:
Ah, that makes sense! Would you mind adding a comment to that effect so future us isn't also confused? Thanks!